### PR TITLE
Correct property access of `expense` in `TransactionItem.js`

### PR DESCRIPTION
**Pull request by SudharakaP**
Related to https://sentry.io/organizations/open-collective/issues/3017053982/?project=1736806&referrer=slack. 

This is a regression introduced at https://github.com/opencollective/opencollective/issues/4691

### CODE REVIEW

**Review by Betree — COMMENTED** [review]
Regression introduced in https://github.com/opencollective/opencollective-frontend/pull/7391 (https://github.com/opencollective/opencollective/issues/4691 is the project specs). No blocker for me but I have not tested the fix and have very little context about this project. Feel free to seek additional feedback and testing as you see fit.

**Comment by SudharakaP** [comment]
@znarf: Assigning this to you for review. This is a regression introduced in https://github.com/opencollective/opencollective-frontend/pull/7391, where I tried to check for `kind` instead since I wanted to make things consistent everywhere (we already use [`kind` check within the `getDisplayedAmount` function](https://github.com/opencollective/opencollective-frontend/blob/26b6fa22b601c59667d2e074937421ff8e58b72a/components/transactions/TransactionItem.js#L46)). However this is a mistake and we need to revert this and check for `expense != null` as otherwise as shown in Sentry we encounter a null pointer.

**Comment by znarf** [comment]
This doesn't look like a proper way to fix it.

Have you investigated why would `expense` be null while `kind === TransactionKind.EXPENSE`?

**Comment by SudharakaP** [comment]
> This doesn't look like a proper way to fix it.
> 
> Have you investigated why would `expense` be null while `kind === TransactionKind.EXPENSE`?

I've recreated this problem in my local. Basically, the situation happens when we refund and expense -> reject it and then delete the expense. So basically since it's deleted the `expense` isn't returned here; but on the users profile the refunded transaction and the refund transaction is shown (and those transaction objects doesn't have the `expense`). My understanding is that this is expected behavior; we should show the refunds and refunded transactions of deleted expenses, but we should not return all deleted expenses and show the details of those. In this case show show the expense tag; 

image

Basically what I was thinking is that we should not return deleted expenses in these cases and thus my fix to not depend `transaction.kind` to decide if the original expense exists. Let me know if there's anything wrong with this reasoning. 🤔

**Comment by znarf** [comment]
I would prefer a solution that keeps showing details in this case.

**Comment by znarf** [comment]
I implemented an alternative fix there https://github.com/opencollective/opencollective-frontend/pull/7500
